ufo2fontir/src/toir.rs

```diff
@@ -24,27 +25,30 @@ pub(crate) fn to_ir_location(loc: &[Dimension]) -> UserSpaceLocation {
         .collect()
 }
 
-pub fn designspace_to_ir(designspace: DesignSpaceDocument) -> Result<Vec<ir::Axis>, Error> {
-    // Truly we have done something amazing here today
-    let ir_axes: Vec<ir::Axis> = designspace.axes.into_iter().map(to_ir_axis).collect();
-
-    // Someday we will return something useful! But ... not today.
-    Ok(ir_axes)
+fn to_user(design_to_user: &PiecewiseLinearMap, design_coord: f32) -> UserSpaceCoord {
+    UserSpaceCoord::new(design_to_user.map(OrderedFloat(design_coord)))
 }
 
-fn design_to_user(value: f32) -> UserSpaceCoord {
-    let coord = DesignSpaceCoord::new(OrderedFloat(value));
-    temporary_design_to_user_conversion(coord)
-}
+pub fn to_ir_axis(axis: &designspace::Axis) -> ir::Axis {
+    // We're not in designspace anymore Dorothy
+    let design_to_user = if let Some(mappings) = &axis.map {
+        let map = mappings
+            .iter()
+            .map(|m| (OrderedFloat(m.input), OrderedFloat(m.output)))
+            .collect();
+        PiecewiseLinearMap::new(map)
+    } else {
+        PiecewiseLinearMap::nop()
+    };
 
-fn to_ir_axis(axis: designspace::Axis) -> ir::Axis {
     ir::Axis {
-        name: axis.name,
-        tag: axis.tag,
-        min: design_to_user(axis.minimum.expect("Discrete axes not supported yet")),
-        default: design_to_user(axis.default),
-        max: design_to_user(axis.maximum.expect("Discrete axes not supported yet")),
+        name: axis.name.clone(),
+        tag: axis.tag.clone(),
         hidden: axis.hidden,
+        min: to_user(&design_to_user, axis.minimum.unwrap()),
```

Comment:
in DS, axis minimum/default/maximum are already in user-space coordinates so no mapping is required here

Reply:
Fixed, ty